packages/sdk/src/marketplaces/LooksRare.ts

```diff
@@ -0,0 +1,573 @@
+    }
+
+    if (quoteAsset) {
+      const quoteAssetPrice = quoteAsset.amount.toString();
+      query.currency = quoteAsset.contractAddress;
+      query.price = { min: quoteAssetPrice, max: quoteAssetPrice };
+    }
+
+    const result = await this._fetchOrders(query);
+
+    return result;
+  }
+
+  /**
+   * Cancels order on LooksRare exchange contract.
+   */
+  async cancelOrder(order: MakeOrderResult): Promise<ContractReceipt> {
+    return this._getExchangeContract().cancelMultipleMakerOrders([order.nonce]);
+  }
+
+  /**
+   * Gets looksrare exchange contract instance.
+   */
+  private _getExchangeContract(): Contract {
+    const exchangeAddress = addressesByNetwork[this.chainId].EXCHANGE;
+
+    return new Contract(exchangeAddress, LooksRareExchangeAbi, this.signer);
+  }
+
+  /**
+   * Gets nonce from LooksRare API.
+   */
+  private async _getNonce(address: string) {
+    return fetch(`${this._getApiUrl(ApiPath.orderNonce, { address })}`)
+      .then((res) => res.json())
+      .then(({ success, data, message }: ApiResponse<string>) => {
+        if (!success) {
+          throw new Error(message);
+        }
+
+        if (!data) {
+          throw new Error("missing data");
+        }
+
+        return data;
+      });
+  }
+
+  /**
+   * Posts make order to LooksRare API.
+   */
+  private _postOrder(payload: {
+    signature: string;
+    tokenId: string;
+    collection: string;
+    strategy: string;
+    currency: string;
+    signer: string;
+    isOrderAsk: boolean;
+    nonce: string;
+    amount: string;
+    price: string;
+    startTime: number;
+    endTime: number;
+    minPercentageToAsk: number;
+    params: unknown[];
+  }): Promise<MakeOrderResult> {
+    const apiKeyHeader: { "X-Looks-Api-Key": string } | {} = this.apiKey
+      ? {
+          "X-Looks-Api-Key": this.apiKey,
+        }
+      : {};
+
+    return fetch(this._getApiUrl(ApiPath.orders), {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        ...apiKeyHeader,
+      },
+      body: JSON.stringify(payload),
+    })
+      .then((res) => res.json())
+      .then(({ success, data, message }: MakeOrderResponse) => {
+        if (!success) {
+          throw new Error(message);
+        }
+
+        if (!data) {
+          throw new Error("missing data");
+        }
+
+        return data;
+      });
+  }
+
+  /**
+   * Fetches order from LooksRare API.
+   */
+  private _fetchOrders(params: FetchOrderParams): Promise<MakeOrderResult[]> {
+    return fetch(this._getApiUrl(ApiPath.orders, parseFetchParams(params)))
+      .then((res) => res.json())
+      .then(({ success, data, message }: GetOrderResponse) => {
+        if (!success) {
+          throw new Error(message);
+        }
+
+        if (!data) {
+          throw new Error("missing data");
+        }
+
+        return data;
+      });
+  }
+
+  /**
+   * Gets LooksRare API origin.
+   */
+  private _getApiUrl(
```

Comment:
It seems that your code editor is missing something regarding eslint support, because you should be getting this error for methods without return value:
`Missing return type on function.eslint[@typescript-eslint/explicit-function-return-type](https://typescript-eslint.io/rules/explicit-function-return-type)`